### PR TITLE
Documentation app

### PR DESCRIPTION
**Pull request by krunalm**
1. created basic code structure for documentation
2. created example of choicelist component from engage-ui.

### CODE REVIEW

File: example/src/Routes.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import App from './containers/App';
+import HomePage from './containers/HomePage';
+import ComponentsPage from './containers/ComponentsPage';
+import NotFoundPage from './containers/NotFoundPage';
+
+export default (
+  <Route path="/" component={App}>
+    <IndexRoute component={HomePage} />
+    <Route path="/component/:component" component={ComponentsPage} />
+    <Route path="*" component={NotFoundPage}/> */
```
Comment (mkulinski) [inline example/src/Routes.tsx]: What is the */ at the end of the line for?

File: example/src/Types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// import * as React from 'react';
```
Comment (mkulinski) [inline example/src/Types.ts]: remove comment.

File: example/src/containers/ComponentsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+export interface IProps {
+  state: IDocument[];
+  actions: typeof actions.requestComponents;
+  routeParams: any;
+}
+
+class ComponentsPage extends React.Component<IProps, any> {
+  render() {
+    const currentComponent = this.props.routeParams.component;
+    const currentState = this.props.state.find((c) => { return c.id === currentComponent; });
```
Comment (mkulinski) [inline example/src/containers/ComponentsPage.tsx]: avoid using single letter parameters whenever possible. Just makes it a little easier for someone that is unfamiliar with the app to see what's going on.

File: example/src/reducers/ComponentReducer.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import * as types from '../constants/ComponentActionTypes';
+import InitialState from './InitialState';
+import { IDocument, IAction } from '../Types';
+
+export default function componentReducer(state: IDocument[] = InitialState.components, action: IAction) {
+  switch (action.type) {
+    case types.REQUEST_COMPONENTS: {
+      return {
+        ...state,
+        // error: false,
```
Comment (mkulinski) [inline example/src/reducers/ComponentReducer.ts]: were you keeping the commented out components in here for when we need to start doing calls to an api?

File: webpack.dev.config.js [hunk 5]
```diff
@@ -10,11 +10,52 @@ BaseConfig.entry = [
...
+BaseConfig.devServer = {
+  historyApiFallback: true,
 };
 
 BaseConfig.plugins.push(new HtmlWebpackPlugin({
   filename: 'index.html',
   template: './example/src/index.html',
 }));
+/*
+BaseConfig.module.rules.splice(2, 1, {
```
Comment (mkulinski) [inline webpack.dev.config.js]: Was the the temp fix for the build issues you were having last week?

File: example/src/Routes.tsx (hunk 1 above)
Comment (krunalm) [inline example/src/Routes.tsx]: @mkulinski Good catch. I was trying something by commenting code and didnt realize it at the end. 
BTW, typescript was compiling successfully even though for this issue. weird stuff.

File: example/src/reducers/ComponentReducer.ts (hunk 4 above)
Comment (krunalm) [inline example/src/reducers/ComponentReducer.ts]: @mkulinski Yes. Now, i think we don't need it anymore. Later we can add it when we actually require it. Thanks for asking.

File: webpack.dev.config.js (hunk 5 above)
Comment (krunalm) [inline webpack.dev.config.js]: @mkulinski Yes. Now, we have proper resolution of that error, I removed comments.

**Comment by krunalm** [comment]
@mkulinski I've resolved conflicts and merged master branch for travis issues.

**Pull request merged**